src/aks-preview/azext_aks_preview/custom.py

```diff
@@ -4383,3 +4387,195 @@ def aks_bastion(cmd, client, resource_group_name, name, bastion=None, port=None,
             )
         finally:
             aks_batsion_clean_up()
+
+
+def aks_agent(
+        cmd,
+        client,
+        resource_group_name,
+        name,
+        prompt,
+        max_steps,
+        config_file,
+        no_interactive=False,
+        no_echo_request=False,
+        show_tool_output=False,
+        refresh_toolsets=False,
+):
+    '''
+    Interact with the AKS agent using a prompt or piped input.
+
+    :param prompt: The prompt to send to the agent.
+    :type prompt: str
+    :param interactive: Whether to run in interactive mode.
+    :type interactive: bool
+    :param max_steps: Maximum number of steps to take.
+    :type max_steps: int
+    :param config_file: Path to the config file.
+    :type config_file: str
+    :param no_interactive: Disable interactive mode.
+    :type no_interactive: bool
+    :param no_echo_request: Disable echoing back the question provided to AKS Agent in the output.
+    :type no_echo_request: bool
+    :param show_tool_output: Whether to show tool output.
+    :type show_tool_output: bool
+    :param refresh_toolsets: Refresh the toolsets status.
+    :type refresh_toolsets: bool
+    '''
+
+    # reverse the value of the variables so that
```

Comment:
if possible, please move your detailed implementation to a subfolder and import it here